schema/overview.md

```diff
@@ -0,0 +1,68 @@
+IPLD Schemas Overview
+=====================
+
+
+Goals
+-----
+
+1. Provide a reasonable, easy to use, general type system for declaring
+  useful properties about data.
+2. Compose nicely over IPLD Data Model.  Schemas should be usable with any
+  data format you can build an IPLD Codec for.
+3. Be efficient to verify (e.g. predictable; roughly linear in the size of
+  the data and schema; and absolutely not turing complete).
+4. Be language-agnostic: many compatible implementations of the schema
+  checker tools should exist, as well as bindings for every language.
+5. Assist rather than obstruct migration.  (We expect data to exist from
+  *before* the current schemas; we need to work well on this case.)
+6. Be a home for collaboration and documentation!  Schemas should provide a
+  natural place for API documentation, and be a reasonable design literature.
+
+In total: **Make developing decentralized & distributed applications
+easier, clearer, and more robust!**
+
+
+Features
+--------
+
+Well-understood basics (sum types, product types, some specific recursive types):
+
+- typed maps (e.g. `{Foo:Bar}`)
+- typed arrays (e.g. `[Foo]`)
```

Comment:
we've so far avoided using "array" and stuck to "list" I think, maybe it'd be safer to keep it that way, or at least put "list" primary, either "typed lists" or "typed lists / arrays"